arduino/smartcar/smartcar.ino

```diff
@@ -335,32 +338,34 @@ void connectivityLoss(){
 }
 
 /**
- * @param sensor : takes the name of the sensor to be checked  as a String.
- * @Returns true if sensor does not detect an obstacle or if it is too close to be detected. False if an obstacle is detected within the range of the sensor.
+ * @param sensor : receives the number of the sensor to be checked. Sensor numbers declared as constant class variables.
+ * @Returns true if sensor does not detect an obstacle or if it is too close to be detected.
+ * False if an obstacle is detected within the range of the sensor.
  */
-bool isClear(String sensor)
-{
-    if (sensor == "frontUS"){
-        return (frontUS.getDistance() == NO_OBSTACLE_VALUE);
-    } else if(sensor == "frontIR") {
-        return (frontIR.getDistance() == NO_OBSTACLE_VALUE);
-    }else if(sensor == "backIR"){
-        return (backIR.getDistance() == NO_OBSTACLE_VALUE);
-    }else if(sensor == "rightIR"){
-        return (rightIR.getDistance() == NO_OBSTACLE_VALUE);
-    }else if(sensor == "leftIR"){
-        return (leftIR.getDistance() == NO_OBSTACLE_VALUE);
-    }else{
-        return false;
+bool isClear(int sensor){
+
+    switch (sensor) {
+        case 1:
+            return (frontUSSensor.getDistance() == NO_OBSTACLE_VALUE);
+        case 2:
+            return (frontIRSensor.getDistance() == NO_OBSTACLE_VALUE);
+        case 3:
+            return (backIRSensor.getDistance() == NO_OBSTACLE_VALUE);
+        case 4:
+            return (rightIRSensor.getDistance() == NO_OBSTACLE_VALUE);
+        case 5:
+            return (leftIRSensor.getDistance() == NO_OBSTACLE_VALUE);
+        default:
+            return false;
```

Comment:
Nice having changed this for a switch. I remember switch were not working with strings in c++ but here you are having numbers so that's good !